app/code/Magento/Catalog/Controller/Adminhtml/Product/Action/Attribute/Save.php

```diff
@@ -162,7 +162,7 @@ public function execute()
                         $this->attributeHelper->getStoreWebsiteId($storeId)
                     );
                     if (!$stockItemDo->getProductId()) {
-                        $inventoryData[] = $productId;
+                        $inventoryData['product_id'] = $productId;
```

Comment:
Hmm, looks like it never worked: https://github.com/magento/magento2/blob/c8b7223139366e8030b0b8ae43406495e46ec95e/lib/internal/Magento/Framework/Service/Data/AbstractObjectBuilder.php#L100